internal/transport/keepalive_test.go

```diff
@@ -749,3 +654,45 @@ func (s) TestTCPUserTimeout(t *testing.T) {
 		}
 	}
 }
+
+func pollForHealthyStream(client *http2Client) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	var err error
+	for ctx.Err() == nil {
+		if _, err = client.NewStream(ctx, &CallHdr{}); err == nil {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
+	return err
+}
+
+func pollForStreamCreationError(client *http2Client) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	for ctx.Err() == nil {
+		if _, err := client.NewStream(ctx, &CallHdr{}); err != nil {
+			break
+		}
+		time.Sleep(50 * time.Millisecond)
+	}
+	if ctx.Err() != nil {
+		return fmt.Errorf("client creation did not fail before test time out")
+	}
+	return nil
+}
+
+func waitForGoAwayTooManyPings(client *http2Client) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	select {
+	case <-client.Error():
+		if reason, _ := client.GetGoAwayReason(); reason != GoAwayTooManyPings {
+			return fmt.Errorf("goAwayReason is %v, want %v", reason, GoAwayTooManyPings)
+		}
+	case <-ctx.Done():
+		return fmt.Errorf("test timed out before getting GoAway with reason:GoAwayTooManyPings from server")
+	}
+	return pollForStreamCreationError(client)
```

Comment:
Either do nothing and let the caller call `pollForStreamCreationError` or just try 1 RPC here (which should always error since the goaway happened)?